kip-0017.md

```diff
@@ -8,105 +8,422 @@ Category: Interface
+  "jsonrpc": "2.0",
+  "result": [
+    {
+      "accounts": [
+        {
+          "account": "kadena:mainnet01:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
+          "publicKey": "38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
+          "kadenaAccounts": [
+            {
+              "name": "w:aoriestnaoirsetnaorisetn",
+              "contract": "coin",
+              "chains": ["0", "1"]
+            },
+            {
+              "name": "k:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
+              "contract": "coin",
+              "chains": [
+                "0",
+                "1",
+                "2",
+                "3",
+                "4",
+                "5",
+                "6",
+                "7",
+                "8",
+                "9",
+                "10",
+                "11",
+                "12",
+                "13",
+                "14",
+                "15",
+                "16",
+                "17",
+                "18",
+                "19"
+              ]
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+#### kadena_getAccounts_v1 Request
+
+This method needs to be called with a `params` object with the following properties:
+
+1. `params`: `Object`\
+    1.1. `accounts`: `Array<Object>`\
+    __ 1.1.1 `account`: `string` - the public key (a Wallet Connect [Account](#wallet-connect-accounts)) for which the Kadena account name is requested.\
+    __ 1.1.2 `contracts`: `Array<string>` (optional) - array of fungible contracts for which the Kadena account name is requested. Returns all known Kadena accounts if omitted.
+
+#### kadena_getAccounts_v1 Response
+
+The method response contains a `result` object that contains the following properties:
+
+1. `result`: `Object`\
+    1.1. `accounts`: `Array<Object>`\
+    __ 1.1.1 `account`: `string` - the Wallet Connect [Account](#wallet-connect-accounts) requested.\
+    __ 1.1.2 `publicKey`: `string` - the public key extracted from the Wallet Connect account requested.\
+    __ 1.1.3 `kadenaAccounts`: `Array<Object>`\
+    ____ 1.1.3.1 `name`: `string` - the account name as stored in the blockchain for the specified contract.\
+    ____ 1.1.3.2 `contract`: `string` - the contract that this account is used for.\
+    ____ 1.1.3.3 `chains`: `Array<string>` - the chains on which this account is present.
+
+> **NB**: The accounts that are returned are the ones that are available in the
+> scope of the wallet (i.e. only the accounts the wallet knows about).
+
+> **NB**: The same Kadena account name can be used in different contracts
+> (e.g. `coin`, `my-fungible`). They require different entries
+> since the chains on which the accounts are present can differ.
+
+> **NB**: We choose to not add chains to the [getAccounts_v1](#kadena_getAccounts_v1-Request) Request, since it would
+> add complexity to the interface. The dApp can filter on the chains needed.
```

Comment:
@jermaine150 thoughts on moving this section as an NB in the Request section?
This also reminded me that it might be useful to add a sentence in the description of method that states something along the lines of:
"Users will known what contracts and chains contain the account name"

Reply:
Yes that's better. I'll move it to the Request section.
That extra description seems a bit trivial to me, I don't think we need it.